language/move-prover/docgen/src/docgen.rs

```diff
@@ -236,9 +240,37 @@ impl<'env> Docgen<'env> {
         }
     }
 
-    /// Generate documentation, returning a pair of output file names and generated content per
-    /// file.
-    pub fn gen(mut self) -> Vec<(String, String)> {
+    /// Generate documentation.
+    pub fn generate(self) -> anyhow::Result<()> {
+        if self.options.include_diagrams {
```

Comment:
I think you come to a much cleaner design if you include the call to dot/img generation here and do not use the diagen crate for this. You do not need to scan files for the usage relation because this is already calculated here. You do not need to scan directories for .move files. All this is currently very bridle and depends on that this is in the libra tree, and duplicates functionality which we already have.

You can just add a function to generate a diagram for a single module here, it should be rather simple function. Then you call this function at the same place where the `use` declarations are generated in this module.

Reply:
The newer commit refactored the code based on the comments here.